### DISCUSSION TITLE
Check constraint works only with lowercase

### DISCUSSION THREAD

**Discussion by nico1996it** [hasura/graphql-engine#7848] [Question] [answered]
![telegram-cloud-photo-size-4-6039718136057542266-y](https://user-images.githubusercontent.com/24296003/139076680-34e76f52-9f4e-4787-bd9d-8aa89a9e962a.jpg)

As you can see, it always converts everything in all lowercase format, therefore if you check a column that is not completely lowercase you will get this error

**Comment by rikinsk** [comment, marked as answer]
This is because of the way Postgres treats identifiers. By default all identifiers are lowercased by Postgres if not wrapped with double-quotes. 

Fom the Postgres docs: "Quoting an identifier also makes it case-sensitive, whereas unquoted names are always folded to lower case. For example, the identifiers FOO, foo, and "foo" are considered the same by PostgreSQL, but "Foo" and "FOO" are different from these three and each other."

For your case, updating the constraint to `"NAME" != null` should fix this.

**Reply by BenoitRanque** [reply]
Please do not use `!=` to compare against `null`, as it will _never_ be true, because null it not equal to itself.
The resulting behavior is that your check constraint will always be true, even when the value is null.

From the [postgres documentation](https://www.postgresql.org/docs/13/functions-comparison.html):
> Do not write expression = NULL because NULL is not “equal to” NULL. (The null value represents an unknown value, and it is not known whether two unknown values are equal.)

Instead, use `"NAME" IS NOT NULL`

Leaving this comment in the hopes it will save someone some headaches in the future
